test/cctest/test_per_process.cc

```diff
@@ -0,0 +1,36 @@
+#include "node_native_module.h"
+
+#include "gtest/gtest.h"
+#include "node_test_fixture.h"
+
+#include <string>
+
+
+using node::native_module::NativeModuleLoader;
+using node::native_module::NativeModuleRecordMap;
+
+class PerProcessTest : public ::testing::Test {
+protected:
+
+  static const NativeModuleRecordMap get_sources_for_test() {
+    return NativeModuleLoader::instance_.source_;
```

Comment:
The builds are failing because this is private. I guess it would also make sense to add a public method to just return the source in a copy, though.

Reply:
I made it a friend
https://github.com/nodejs/node/blob/84aca5ca594f86ce9b6b86e6af039e78f90deaef/src/node_native_module.h#L89
and it compiles & passes with [MSVC](https://ci.nodejs.org/job/node-test-commit-windows-fanned/27662/), now I need to figure out gcc & clang 🤷‍♂

Reply:
Seems like it was a namespace issue
https://github.com/nodejs/node/blob/c178598d309319e9a9e9692d6560a9205cd1b307/src/node_native_module.h#L89